receiver/sapnetweaverreceiver/scraper.go

```diff
@@ -115,34 +114,47 @@ func (s *sapNetweaverScraper) collectAlertTree(_ context.Context, now pcommon.Ti
 		return
 	}
 
+	toggleSwapSpaceFlag := false
```

Comment:
What is this actually used for? At first glance it appears that we are just setting it to true and doesn't impact collection?

Reply:
Nevermind I think I see how it wokrs, had to re-read the loop

Reply:
So there are multiple metrics with the name "Percentage_Used". They however have different parent ID which unfortunately is not consistent number across different instance. The "Percentage_Used" that is desired is the one that comes after "Swap_Space".  This flag here is helps to select a single, correct occurrence of "Percentage_Used" for the memory usage.